src/corelibs/U2View/src/ov_msa/exclude_list/MsaExcludeList.h

```diff
@@ -0,0 +1,201 @@
+
+    void moveMaSelectionToExcludeList(MSAEditor *msaEditor);
+
+    void updateState(MSAEditor *msaEditor);
+
+    MsaExcludeListWidget *findActiveExcludeList(MSAEditor *msaEditor);
+
+    MsaExcludeListWidget *openExcludeList(MSAEditor *msaEditor);
+
+    void updateMsaEditorSplitterStyle(MSAEditor *msaEditor);
+};
+
+/** Stores per step (Undo or Redo) information used by Exclude List to keep synchronized state with MSA Editor during Undo/Redo ops. */
+struct UndoRedoStep {
+    /** Direction of the original move. */
+    bool isMoveFromMsaToExcludeList = false;
+    /** Affected Exclude List row ids. The row ids keeps the same order as MSA rows being changed during the move. */
+    QList<int> excludeListRowIdsDelta;
+};
+
+/**
+ * Exclude list widget component for MsaEditor.
+ * The 'exclude list' renders a list of sequences excluded from the multiple alignment object.
+ *
+ * The widget contains a data model for Exclude List and a limited Undo/Redo support for MSA:
+ * the Undo-Redo is supported only for the life-time of the widget: when Exclude List is closed the Undo/Redo stack of the widget is reset.
+ */
+class MsaExcludeListWidget : public QWidget {
+    Q_OBJECT
+public:
+    MsaExcludeListWidget(QWidget *parent, MSAEditor *msaEditor, MsaExcludeListContext *viewContext);
+
+    /** Triggers auto-save of the current exclude list state. */
+    ~MsaExcludeListWidget();
+
+    /** Moves currently selected MSA rows to the Exclude List. */
+    void moveMsaSelectionToExcludeList();
+
+    /** Moves currently selected Exclude List rows to MSA. */
+    void moveExcludeListSelectionToMaObject();
+
+    /** Tracks MSA object "Save Document" task and auto-saves Exclude List on successfull taks completion. */
+    void trackMsaObjectSaveTask(Task *task);
+
+    /** Returns preferred size hint used on instantiation. This size defines initial widget height in the MSA splitter. */
+    QSize sizeHint() const override;
+
+private:
+    /** Updates state of all child widgets. */
+    void updateState();
+
+    /** Updates currently visible sequence text based on name-list selection. */
+    void updateSequenceView();
+
+    /** Adds a minimal support of the synchronized Undo/Redo for the Exclude List and MSA Editor. */
+    void handleUndoRedoInMsaEditor(const MultipleAlignment &maBefore, const MaModificationInfo &modInfo);
+
+    /** Adds a name list entry. If excludeListRowId <= 0 auto-assigns a new row id. Returns a valid exclude list row id. */
+    int addEntry(const DNASequence &sequence, int excludeListRowId = 0);
+
+    /** Adds a name list entry from MSA row. Fixes non-defined DNA sequence alphabet of MSA row. */
+    int addMsaRowEntry(const MultipleAlignmentRow &row, int excludeListRowId = 0);
+
+    /** Remove sequence entries from the Exclude List model. */
+    void removeEntries(const QList<QListWidgetItem *> &items);
+
+    /** Triggers context menu for the name list. */
+    void showNameListContextMenu(const QPoint &pos);
```

Comment:
Why do you need the argument if it's unused?

Reply:
Because this function is bound as a callback  for the "void customContextMenuRequested(const QPoint &pos)"
Do not have a consistent style in UGENE where to use it and where not.  If you will search for customContextMenuRequested() in UGENE you will find in 90% of cases it has the parameter.

I removed the param to keep moving forward.